test/thorest-api/subscriptions/ws-transfers.test.ts

```diff
@@ -1,41 +1,241 @@
+            { txOrigin: wallet.address },
+            account,
+            wallet,
+        )
 
         expect(relevantEvent).not.toBeUndefined()
         expect(relevantEvent?.meta?.txOrigin).toEqual(sender)
+        expect(relevantEvent?.recipient).toEqual(account)
+    })
+
+    it.e2eTest(
+        'should work for valid txOrigin different than sender',
+        'all',
+        async () => {
+            const wallet: ThorWallet = ThorWallet.withFunds()
+            const simpleTransfer = await wallet.deployContract(
+                SimpleTransfer.bytecode,
+                SimpleTransfer.abi,
+            )
+            const sender = simpleTransfer.address // contract that sends VET
+            const txOrigin = wallet.address // EOA that triggered contract call
+            const account = generateAddress()
+
+            const functionData =
+                SimpleTransfer.createInterface().encodeFunctionData(
+                    'transfer',
+                    [account, 1],
+                )
+
+            const events: components['schemas']['SubscriptionTransferResponse'][] =
+                []
+
+            Client.raw.subscribeToTransfers({ sender: sender }, (event) => {
+                events.push(event)
+            })
+
+            const receipt = await wallet.sendClauses(
+                [
+                    {
+                        to: sender,
+                        value: '0x1',
+                        data: functionData,
+                    },
+                ],
+                true,
+            )
+
+            expect(receipt).toBeDefined()
+            expect(receipt!.reverted).toBe(false)
+
+            // Sleep for 1 sec to ensure the beat is received
+            await new Promise((resolve) => setTimeout(resolve, 1000))
+
+            const relevantEvent = events.find((event) => {
+                return event.meta?.txID === receipt.meta?.txID
+            })
+
+            expect(relevantEvent).not.toBeUndefined()
+            expect(relevantEvent?.meta?.txOrigin).not.toEqual(sender)
+            expect(relevantEvent?.meta?.txOrigin).toEqual(txOrigin)
+        },
+    )
+
+    it.e2eTest('should get 2 notifications', 'all', async () => {
+        const transferTxs: { id: string }[] = []
+        const txpoolTxs: { id: string }[] = []
+
+        const account = generateAddress()
+
+        Client.raw.subscribeToTransfers({ recipient: account }, (event) => {
+            const txID = event.meta?.txID
+            if (txID) {
+                transferTxs.push({ id: txID })
+            }
+        })
+
+        Client.raw.subscribeToTxPool((txId) => {
+            txpoolTxs.push(txId)
+        })
+
+        // Sleep for 1 sec to ensure the beat is received
+        await new Promise((resolve) => setTimeout(resolve, 1000))
+
+        expect(transferTxs).not.toBeUndefined()
+        expect(txpoolTxs).not.toBeUndefined()
+        expect(transferTxs).toEqual(txpoolTxs)
```

Comment:
@MakisChristou looks like this test was retried twice, pointing to instability. (see below)

FYI @leszek-vechain